pandas/tests/indexes/test_indexing.py

```diff
@@ -248,6 +249,20 @@ def test_putmask_with_wrong_mask(self, index):
             index.putmask("foo", fill)
 
 
+class TestCasting:
+    def test_maybe_cast_with_dtype(self):
```

Comment:
This test is better suited in `pandas/tests/indexes/base_class/test_where.py`

Reply:
The main issue was the casting so I was not sure, I could move it anyway.

Reply:
Thanks, Generally we group tests by the method calls on the particular objects (`Index.where`)